### ISSUE TITLE
Toy example for PPO doesn't learn as much as expected

### ISSUE THREAD

**Issue by trialan** [CarperAI/trlx#214]
### 🐛 Describe the bug

I'm running a toy example to check that PPO behaves as I'd expect. The setup is this: dataset is just prompts which say "Hi my name" and I want the model to always generate the next token to be " is". I would expect this to be very easy for the model to learn. Here are [the WandB plots](https://wandb.ai/chaiml/ppo-runs/runs/2mg8mi7p?workspace=user-tr416).

Instead of quickly jumping from a reward of 0.67 to a reward of 1, it goes to about 0.8 and doesn't manage to go higher. All the params I'm using are visible in the WandB overview, at high-level: it's gpt-neo-125M with init_kl_coef of 0.03 and vf_coef=0.1. ([Same thing happens with gpt-neo-1.3B](https://wandb.ai/chaiml/ppo-runs/runs/65y9w8yd?workspace=user-tr416))

Is this a bug or am I using the code wrong?

### Additional system and package information

transformers: 4.20.1

**Comment by pruksmhc** [comment]
following since I've also been having issues with training with PPO

**Comment by LouisCastricato** [comment]
The examples in the read me do not actually work, they're just there for reference. Using PPO to get a model to generate a single word is quite hard. cc @reciprocated  let's update the examples in the read me.

**Comment by reciprocated** [comment]
Hey @trialan, thanks for doing those experiments! @LouisCastricato This toy problem is much easier than the one in the readme since it already starts with positive reward (in fact almost mean 1 in case of 1.3B) but eventually collapses back to 0, which I'd say is probably indicative of some bug almost regardless of exact hyperparameters used. If one treats this problem as a multi-armed bandit, generating a single word should be as easy as doing fine-tuning, especially if that arm/word is already found. I'll try to debug this, and if you could share the code you used that would very be helpful. Thanks!

**Comment by reciprocated** [comment]
It seems that it is easy for PPO to solve this task while using both using default config values from `ppo_config.yml` as well as your suggested changes:
https://wandb.ai/sorry/trlx/reports/Toy-example-for-PPO-does-learn-as-much-as-expected---VmlldzozNDczOTU0

Those runs were done using the following script and you can reproduce them on the current [main](https://github.com/CarperAI/trlx/tree/b5addd48e258f7945a59ba58e30cfe55e6a0c93e)
```python
import pathlib

import yaml

import trlx
from trlx.data.configs import TRLConfig

config_path = pathlib.Path(__file__).parent.joinpath("../configs/ppo_config.yml")
with config_path.open() as f:
    default_config = yaml.safe_load(f)


def main(hparams={}):
    config = TRLConfig.update(default_config, hparams)
    config.train.num_epochs = 10
    config.method.gen_kwargs["max_new_tokens"] = 1
    config.tokenizer.tokenizer_path = config.model.model_path = "EleutherAI/gpt-neo-125M"

    prompts = ["Hi my name"] * 128

    trlx.train(
        reward_fn=lambda samples, **kwargs: [s.endswith("is") for s in samples],
        prompts=prompts,
        eval_prompts=prompts,
        config=config,
    )


if __name__ == "__main__":
    main()
```
I'm going to have to close this issue, however don't hesitate to re-open or open a new one if you have some further questions or problems
